int32 SUCCESS=0
int32 FAILURE=1
int32 CRITICAL_FAILURE=2
int32 result
